tests/helpers/test_discovery.py

```diff
@@ -129,7 +130,9 @@ def test_circular_import(self):
 
         def component_setup(hass, config):
             """Set up mock component."""
-            discovery.load_platform(hass, "switch", "test_circular", "disc", config)
+            discovery.load_platform(
+                hass, Platform.SWITCH, "test_circular", "disc", config
```

Comment:
`"disc"` is not a valid type for the discovery info. Should we fix that now or later?

Reply:
I have adjusted - just need CI to run now (tests pass on my machine)